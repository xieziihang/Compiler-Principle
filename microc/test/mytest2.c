void main()
{
  int i;
  print i;
  println;
  i++;
  print i;
  println;
  i--;
  print i;
  println;
  int j;
  j = 3;
  print j;
  ++j;
  println;
  print j;
  println;
  --j;
  print j;
}