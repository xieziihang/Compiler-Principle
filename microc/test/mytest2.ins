LDARGS 0
CALL (0, "L1_main")
STOP
FLabel (0, "L1_main")
INCSP 1
GETBP
OFFSET 0
ADD
CSTI 0
STI
INCSP -1
GETBP
OFFSET 0
ADD
LDI
PRINTI
INCSP -1
CSTI 10
PRINTC
INCSP -1
GETBP
OFFSET 0
ADD
DUP
LDI
CSTI 1
ADD
STI
INCSP -1
GETBP
OFFSET 0
ADD
LDI
PRINTI
INCSP -1
CSTI 10
PRINTC
INCSP -1
GETBP
OFFSET 0
ADD
DUP
LDI
CSTI 1
SUB
STI
INCSP -1
GETBP
OFFSET 0
ADD
LDI
PRINTI
INCSP -1
CSTI 10
PRINTC
INCSP -1
INCSP 1
GETBP
OFFSET 1
ADD
CSTI 3
STI
INCSP -1
GETBP
OFFSET 1
ADD
LDI
PRINTI
INCSP -1
GETBP
OFFSET 1
ADD
DUP
LDI
CSTI 1
ADD
STI
INCSP -1
CSTI 10
PRINTC
INCSP -1
GETBP
OFFSET 1
ADD
LDI
PRINTI
INCSP -1
CSTI 10
PRINTC
INCSP -1
GETBP
OFFSET 1
ADD
DUP
LDI
CSTI 1
SUB
STI
INCSP -1
GETBP
OFFSET 1
ADD
LDI
PRINTI
INCSP -1
INCSP -2
RET -1