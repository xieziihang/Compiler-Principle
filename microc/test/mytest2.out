24 19 0 5 25 15 1 13 0 0 1 0 0 12 15 -1 13 0 0 1 11 22 15 -1 0 10 23 15 -1 13 0 0 1 9 11 0 1 1 12 15 -1 13 0 0 1 11 22 15 -1 0 10 23 15 -1 13 0 0 1 9 11 0 1 2 12 15 -1 13 0 0 1 11 22 15 -1 0 10 23 15 -1 15 1 13 0 1 1 0 3 12 15 -1 13 0 1 1 11 22 15 -1 13 0 1 1 9 11 0 1 1 12 15 -1 0 10 23 15 -1 13 0 1 1 11 22 15 -1 0 10 23 15 -1 13 0 1 1 9 11 0 1 2 12 15 -1 13 0 1 1 11 22 15 -1 15 -2 21 -1